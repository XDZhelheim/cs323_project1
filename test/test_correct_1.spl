int test_correct_1(int a)
{
  int b = 'd';
  a = 'c';
  while (a < b) 
  {
    a = a + 1;
  }
}