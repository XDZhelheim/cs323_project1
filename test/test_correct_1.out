Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test_correct_1
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: b
                ASSIGN
                Exp (3)
                  CHAR: 'd'
            SEMI
        StmtList (4)
          Stmt (4)
            Exp (4)
              Exp (4)
                ID: a
              ASSIGN
              Exp (4)
                CHAR: 'c'
            SEMI
          StmtList (5)
            Stmt (5)
              WHILE
              LP
              Exp (5)
                Exp (5)
                  ID: a
                LT
                Exp (5)
                  ID: b
              RP
              Stmt (6)
                CompSt (6)
                  LC
                  StmtList (7)
                    Stmt (7)
                      Exp (7)
                        Exp (7)
                          ID: a
                        ASSIGN
                        Exp (7)
                          Exp (7)
                            ID: a
                          PLUS
                          Exp (7)
                            INT: 1
                      SEMI
                  RC
        RC
